HippoMocks/hippomocks.h

```diff
@@ -1256,6 +1263,57 @@ std::pair<int, int> virtual_index(T t)
 
 	if (conv.u.vindex != 0)
 		return std::pair<int, int>((conv.u.delta + conv.u.vtordisp)/sizeof(void*), conv.u.vindex * 2 + 1);
+	
+#elif defined (__SUNPRO_CC)
+    union {
+        T t;
+        struct
+        {
+            unsigned char *value;
+            unsigned long baseoffs;
+        } u;
+    } conv;
+    conv.t = t;
+   //is virtual call
+    unsigned char prolog[]= {0x55, 0x8b, 0xec, 0x83, 0xec,
+            0x04, 0x89, 0x5d, 0xfc,
+            0x8b, 0x45};//, 0x08, 0x8b, 0x40, 0x00,
+            //0x8b, 0x40}; //movl <index>
+    unsigned char* opcodes = (unsigned char*)(*(unsigned int*)&t);
```

Comment:
Instead of this line, change the struct's "conv.u.value" to be an unsigned char**, and use that.